### PR TITLE
Don't pass a steps object if there are no additional steps. 

### LINKED ISSUE #35: Can't be used for templates when template step override is set to false.
https://github.com/transloadit/java-sdk/blob/69494ef7b79ced5ac2edb509e23dddd987467e92/src/main/java/com/transloadit/sdk/Assembly.java#L36

This line creates a new empty steps object. This affects any of our libraries/classes that extend this. So,
- https://github.com/transloadit/java-sdk/blob/69494ef7b79ced5ac2edb509e23dddd987467e92/src/main/java/com/transloadit/sdk/async/AsyncAssembly.java
- https://github.com/transloadit/android-sdk/blob/master/transloadit-android/src/main/java/com/transloadit/android/sdk/AndroidAsyncAssembly.java

I believe(will verify) that the empty steps object is being converted to an actual steps object in JSON that is sent to our underlying API.

### PR DESCRIPTION
**Pull request by ethanwillis**
**Commit Message:**
- Fix passing empty steps objects.
- Bump version 0.1.5 -> 0.1.6 in a bunch of places.

Fixes #35

### CODE REVIEW

File: src/main/java/com/transloadit/sdk/Assembly.java [hunk 1]
```diff
@@ -187,7 +187,9 @@ public void setTusURLStore(TusURLStore store) {
     public AssemblyResponse save(boolean isResumable)
             throws RequestException, LocalOperationException {
         Request request = new Request(getClient());
-        options.put("steps", steps.toMap());
+     		if(!steps.toMap().isEmpty()) {
+					options.put("steps", steps.toMap());
```
Comment (kvz) [inline src/main/java/com/transloadit/sdk/Assembly.java:191]: Indentation looks a bit wild here, but maybe that's just java :)

**Review by kvz — APPROVED** [review]
LGTM, and thank you for contributing this! Needs a check by @Acconut and then we can merge

File: src/main/java/com/transloadit/sdk/Assembly.java (hunk 1 above)
Comment (ethanwillis) [reply src/main/java/com/transloadit/sdk/Assembly.java:191]: Oh, I use tabs and not spaces, that's probably it :)

**Review by Acconut — APPROVED** [review]
Looks good!

**Pull request merged**